src/Validation/Requires.cs

```diff
@@ -189,14 +190,7 @@ public static void NotNullOrEmpty([ValidatedNotNull, NotNull] System.Collections
                 throw new ArgumentNullException(parameterName);
             }
 
-            bool hasElements = false;
-            foreach (object value in values)
-            {
-                hasElements = true;
-                break;
-            }
-
-            if (!hasElements)
+            if (!values.GetEnumerator().MoveNext())
```

Comment:
The old code would try casting the `IEnumerator` to `IDisposable` and dispose it if it did. It would do this in a try/finally to ensure it happened. We lost that in this method. Can you add it back?

Reply:
I can make this change, but I double-checked the code and didn't see anywhere that had it wrapped.

Reply:
I'm not sure what you mean. The try finally behavior I'm referring to comes automatically from the compiler by virtue of the foreach loop.

Reply:
LOL. Oh! I thought you mean that it was explicitly wrapped in a try..finally. 🤦

Reply:
I didn't think to wrap it initially because `IEnumerator` doesn't implement `IDisposable`, although `IEnumerator<T>` does.